source/utils/path.cpp

```diff
@@ -230,13 +231,14 @@ const std::string &path::string() const
 std::wstring path::wstring() const
 {
     const std::string &path_str = string();
-    int size = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, path_str.c_str(), static_cast<int>(path_str.length()), nullptr, 0);
+    const int allocated_size = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, path_str.c_str(), static_cast<int>(path_str.length()), nullptr, 0);
 
-    if (size > 0)
+    if (allocated_size > 0)
     {
-        std::wstring path_converted(size, L'\0');
+        std::wstring path_converted(allocated_size, L'\0');
         // Note: const_cast is NOT necessary in C++17 and newer!
-        size = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, path_str.c_str(), static_cast<int>(path_str.length()), const_cast<wchar_t *>(path_converted.data()), size);
+        const int actual_size = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, path_str.c_str(), static_cast<int>(path_str.length()), &path_converted.at(0), allocated_size);
```

Comment:
Will an empty input path now generate an exception? Maybe we should add some unit tests.

Reply:
No, because I'm explicitly checking whether the string would result in an empty string `if (allocated_size > 0)`. There's no reason that an empty path would result in a non-empty converted string. Anyway, I added a unit test nevertheless in my newest commit.

Reply:
By the way, this is yet another reason why I prefer using `data()` in C++17 and newer (instead of `&path_converted[0]` or something similar), because such questions don't have to be asked (as long as the string/vector length is correct, obviously).